hugegraph-core/src/main/java/com/baidu/hugegraph/auth/StandardAuthManager.java

```diff
@@ -433,10 +451,239 @@ public RolePermission loginUser(String username, String password) {
+            } else {
+                E.checkArgument(!graphs.contains(graph),
+                                "Graph has included in " +
+                                "project that id is %s",
+                                id);
+            }
+            graphs.add(graph);
+            project.graphs(graphs);
+            return this.project.update(project);
+        } finally {
+            locks.unlock();
+        }
+    }
+
+    @Override
+    public Id updateProjectRemoveGraph(Id id, String graph) {
+        E.checkArgument(!Strings.isNullOrEmpty(graph),
+                        "Update project add graph failed, graph "
+                        + "not be null or empty");
+
+        LockUtil.Locks locks = new LockUtil.Locks(this.graph.name());
+        try {
+            locks.lockWrites(LockUtil.VERTEX_LABEL_ADD_UPDATE, id);
+
+            HugeProject project = this.project.get(id);
+            E.checkArgumentNotNull(project,
+                                   "Project that id is %s is not found",
+                                   id.asString());
+            Set<String> graphs = project.graphs();
+            if (graphs == null || !graphs.contains(graph)) {
+                return id;
+            }
+            graphs.remove(graph);
+            project.graphs(graphs);
+            return this.project.update(project);
+        } finally {
+            locks.unlock();
+        }
+    }
+
+    @Override
+    public HugeProject getProject(Id id) {
+        return this.project.get(id);
+    }
+
+    @Override
+    public List<HugeProject> listAllProject(long limit) {
+        return this.project.list(limit);
+    }
+
     /**
      * Maybe can define an proxy class to choose forward or call local
      */
     public static boolean isLocal(AuthManager authManager) {
         return authManager instanceof StandardAuthManager;
     }
+
+    public <R> R commit(Callable<R> callable) {
+        this.groups.shouldCommitTrans(false);
+        this.access.shouldCommitTrans(false);
+        this.targets.shouldCommitTrans(false);
+        this.project.shouldCommitTrans(false);
+        this.belong.shouldCommitTrans(false);
+        this.users.shouldCommitTrans(false);
+
+        try {
+            R result = callable.call();
+            this.graph.systemTransaction().commit();
+            return result;
+        } catch (Throwable throwable) {
+            this.groups.shouldCommitTrans(true);
+            this.access.shouldCommitTrans(true);
+            this.targets.shouldCommitTrans(true);
+            this.project.shouldCommitTrans(true);
+            this.belong.shouldCommitTrans(true);
+            this.users.shouldCommitTrans(true);
+            try {
+                this.graph.systemTransaction().rollback();
+            } catch (Throwable rollbackException) {
+                LOG.error("Rollback failed", rollbackException);
+            }
+
+            throw new HugeException("Call failed", throwable);
```

Comment:
rethrow throwable

Reply:
if i rethrow throwable, this method must be signature with exception, if i do it, top logic must try-catch wrap it;
so i throw HugeException that is a runtime exception

Reply:
throw the throwable itself if it is instance of HugeException, else wrap with HugeException, and also improve the error message.